src/tools/perf/cuda/ucp_cuda_kernel.cu

```diff
@@ -159,8 +168,8 @@ private:
 
         for (unsigned i = 0; i < count; ++i) {
             indices[i] = i;
-            lengths[i] = (i == count - 1) ? ONESIDED_SIGNAL_SIZE :
-                                            perf.params.msg_size_list[i];
+            lengths[i] = (has_counter && i == count - 1) ? ONESIDED_SIGNAL_SIZE :
```

Comment:
Maybe move handling `has_counter` outside the loop like you did in `init_mem_list`